keel/quotations/admin.py

```diff
@@ -16,7 +28,38 @@ class QuotationAdmin(admin.ModelAdmin):
 
 
 class QuotationMilestioneAdmin(CustomBaseModelAdmin):
-    pass
+    
+    def save_model(self, request, obj, form, change):
+        id = str(obj.pk)
+        customer = obj.quotation.user
+        
+        try:
+            case = Case.objects.get(user=customer)
```

Comment:
At time of MileStone creation, customer will not have any case so no need to check for it. Just remove this try catch block completely.

In PaymentManager generate_payment_details function pass Null as Case Id.